osu.Game.Rulesets.Osu/Utils/VectorUtils.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Game.Rulesets.Osu.UI;
+using osuTK;
+
+namespace osu.Game.Rulesets.Osu.Utils
+{
+    public static class VectorUtils
```

Comment:
Name feels too general, because while `RotateVectorTowardsVector()` is a pretty generic method, `RotateAwayFromEdge` is pretty osu!-specific (baked in playfield sizes). Maybe `OsuHitObjectGenerationUtils`?